testing/MLDB-724-time-arithmetic.py

```diff
@@ -604,21 +613,21 @@ def test_MLDB_1509(self):
                            {
                                "ts": "2015-01-01T00:00:00Z"
                            },
-                           "2015-01-03T00:00:00Z"
+                           "2015-01-01T00:00:00Z"
```

Comment:
Edit: We lose the *some of the power* of the "WHEN" operator in the particular use case that interrests me.

With the current implementation, that cell/value is kept with a `WHEN value_timestamp() > to_timestamp('2015-01-02T00:00:00Z')`. I want it to be discarded.

I'm investigating alternatives. I can probably push the distinct_timestamps usage later in the pipeline at the cost of  performance.